libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py

```diff
@@ -219,4 +225,9 @@ def load_data(
                 + text_content[span["offset"] + span["length"] :]
             )
 
+        # save the text content into markdown format
+        if self.cache_dir is not None:
+            print(file_name)
+            with open(self.cache_dir / f"{file_name}.md", "w") as f:
```

Comment:
Since self.cache_dir can be string, which will not work with `/` operator, should use `Path(self.cache_dir)` instead.